src/Microsoft.AspNetCore.Mvc.TagHelpers/PartialTagHelper.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Mvc.ViewFeatures.Internal;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    /// <summary>
+    /// Renders a partial view.
+    /// </summary>
+    [HtmlTargetElement("partial", Attributes = "name", TagStructure = TagStructure.WithoutEndTag)]
+    public class PartialTagHelper : TagHelper
+    {
+        private readonly ICompositeViewEngine _viewEngine;
+        private readonly IViewBufferScope _viewBufferScope;
+
+        public PartialTagHelper(
+            ICompositeViewEngine viewEngine,
+            IViewBufferScope viewBufferScope)
+        {
+            _viewEngine = viewEngine ?? throw new ArgumentNullException(nameof(viewEngine));
+            _viewBufferScope = viewBufferScope ?? throw new ArgumentNullException(nameof(viewBufferScope));
+        }
+
+        /// <summary>
+        /// The name of the partial view used to create the HTML markup.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// A model to pass into the partial view.
+        /// </summary>
+        public object Model { get; set; }
+
+        /// <summary>
+        /// A <see cref="ViewDataDictionary"/> to pass into the partial view.
+        /// </summary>
+        public ViewDataDictionary ViewData { get; set; }
+
+        [HtmlAttributeNotBound]
+        [ViewContext]
+        public ViewContext ViewContext { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var viewBuffer = new ViewBuffer(_viewBufferScope, Name, ViewBuffer.PartialViewPageSize);
+            using (var writer = new ViewBufferTextWriter(viewBuffer, Encoding.UTF8))
+            {
+                await RenderPartialViewAsync(writer);
+
+                // Reset the TagName. We don't want `partial` to render.
+                output.TagName = null;
+                output.Content.SetHtmlContent(viewBuffer);
+            }
+        }
+
+        private async Task RenderPartialViewAsync(TextWriter writer)
```

Comment:
Just a suggestion: Could make this tag helper an `HtmlHelper` subclass which implements `ITagHelper`. Would have to make `ViewContext` a `new` property that calls `Contextualize(...)` in its setter but that should be the primary wrinkle.

Reply:
Forgot the punchline: Then could use `RenderPartialCoreAsync(...)`.

Reply:
The sample @DamianEdwards has essentially DIs a `IHtmlHelper`, `Contextualizes` it and then renders the view using it. That kinda felt odd since we don't have any other use cases of doing this in any other tag helper. Given we already have a similarish copy of the code in PartialViewExecutor, I didn't think it was terrible to have yet another copy of it here.